content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md

```diff
@@ -123,19 +123,43 @@ by using a list of well known domain socket paths. To use different container ru
 if there are more than one installed on the provisioned node, specify the `--cri-socket`
 argument to `kubeadm init`. See [Installing runtime](/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#installing-runtime).
 1. (Optional) Unless otherwise specified, kubeadm uses the network interface associated
-with the default gateway to advertise the master's IP. To use a different
+with the default gateway to advertise the control-plane's IP. To use a different
 network interface, specify the `--apiserver-advertise-address=<ip-address>` argument
 to `kubeadm init`. To deploy an IPv6 Kubernetes cluster using IPv6 addressing, you
 must specify an IPv6 address, for example `--apiserver-advertise-address=fd00::101`
 1. (Optional) Run `kubeadm config images pull` prior to `kubeadm init` to verify
-connectivity to gcr.io registries.   
+connectivity to gcr.io registries.
 
-Now run:
+To initialize the control-plane node run:
 
 ```bash
 kubeadm init <args>
 ```
 
+### Considerations about ControlPlaneEndpoint
+
+Instead of using `--apiserver-advertise-address`, `kubeadm init` also allows you to use the `--control-plane-endpoint` flag.
+While `--apiserver-advertise-address` can be used to set the advertise address for this particular control-plane node's API
+server, `--control-plane-endpoint` is preferred and can be used to set the shared endpoint for all control-plane nodes.
+Such an endpoint can be either a DNS name or an IP address of a load-balancer.
+
+If you have plans to upgrade this single control-plane kubeadm cluster to high-availability you should definitely use
+`--control-plane-endpoint`. This will allow you to pass any DNS name and later change the address that the DNS name resolves to.
+Use your preferred solution for mapping this node's IP address to a DNS name. For example:
+
+```
+192.168.0.102 cluster-endpoint
+```
+
+Where `192.168.0.102` is the IP address of this node and `cluster-endpoint` is a custom DNS name that maps to this IP.
+
+This will allow you to pass `--control-plane-endpoint=cluster-endpoint` to `kubeadm init` and pass the same DNS name to `kubeadm join`.
+Later you can modify `cluster-endpoint` to point to the address of your load-balancer in an HA scenario.
+
+Without using this abstraction, `kubeadm init` will sign the API server certificate (SAN) to only allow the advertise address of
```

Comment:
I would simplify and say that 
turning a single master cluster created without a `--control-plane-endpoint` into a highly available cluster is not supported by kubeadm.